hydra/grammar/OverrideLexer.g4

```diff
@@ -43,29 +44,28 @@ BRACE_CLOSE: WS? '}';
 VALUE_COLON: WS? ':' WS? -> type(COLON);
 VALUE_EQUAL: WS? '=' WS? -> type(EQUAL);
 
-// Types
-fragment NZ_DIGIT: [1-9];
-fragment INT_PART: DIGIT+;
-fragment FRACTION: '.' DIGIT+;
-fragment POINT_FLOAT: INT_PART? FRACTION | INT_PART '.';
-fragment EXPONENT: [eE] [+-]? DIGIT+;
-fragment EXPONENT_FLOAT: ( INT_PART | POINT_FLOAT) EXPONENT;
-FLOAT: [-]?(POINT_FLOAT | EXPONENT_FLOAT | [Ii][Nn][Ff] | [Nn][Aa][Nn]);
-INT: [-]? ('0' | (NZ_DIGIT DIGIT*));
+// Numbers.
+
+fragment POINT_FLOAT: INT_UNSIGNED '.' | INT_UNSIGNED? '.' DIGIT (('_')? DIGIT)*;
+fragment EXPONENT_FLOAT: (INT_UNSIGNED | POINT_FLOAT) [eE] [+-]? INT_UNSIGNED;
+FLOAT: [+-]? (POINT_FLOAT | EXPONENT_FLOAT | [Ii][Nn][Ff] | [Nn][Aa][Nn]);
+INT: [+-]? INT_UNSIGNED;
+
+// Other reserved keywords.
 
 BOOL:
       [Tt][Rr][Uu][Ee]      // TRUE
     | [Ff][Aa][Ll][Ss][Ee]; // FALSE
 
 NULL: [Nn][Uu][Ll][Ll];
 
-UNQUOTED_CHAR: [/\-\\+.$*];  // other characters allowed in unquoted strings
-VALUE_ID: ID -> type(ID);
-VALUE_DOT_PATH: DOT_PATH -> type(DOT_PATH);
+UNQUOTED_CHAR: [/\-\\+.$%*];  // other characters allowed in unquoted strings
```

Comment:
Adding `%` which is a valid character in OmegaConf (see https://github.com/omry/omegaconf/blob/9f7d3566c1ce99da1d112e0e4656bf6651b3a62e/omegaconf/_utils.py#L328)

I believe I just forgot it in the initial version.